doc/source/dev/index.rst

```diff
@@ -189,7 +189,7 @@ Stylistic Guidelines
 
    import numpy as np
 
-* For C code, see the :ref:`numpy-c-style-guide<style_guide>`
+* For C code, see `NEP 45 <https://numpy.org/neps/nep-0045-c_style_guide.html>`_.
```

Comment:
If you add `.. _NEP45:` to the top of the nep (see #16450 for example), you could reference it via intersphinx here as well, e.g. `` :ref:`NEP 45 <nep45>` ``